### ISSUE TITLE
onRecognitionComplete triggered twice

### ISSUE THREAD

**Issue by toshiossada** [rxlabz/speech_recognition#73]
When I speak some sentence with speech_recognition  the "onRecognitionComplete" are triggered twice, for example, I said "Hello how are you", it triggered "Hello how are" and "Hello how are you"... it's anyway to fix this and trigger just once??

Following logs bellow

> I/flutter (15202): true
> I/flutter (15202): _platformCallHandler call speech.onSpeechAvailability true
> I/flutter (15202): _platformCallHandler call speech.onRecognitionStarted null
> I/flutter (15202): _platformCallHandler call speech.onSpeech
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello how
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello how are
> I/flutter (15202): _platformCallHandler call speech.onRecognitionComplete Hello how are
> I/flutter (15202): _platformCallHandler call speech.onSpeech Hello how are you
> I/flutter (15202): _platformCallHandler call speech.onRecognitionComplete Hello how are you

I'm using this way 

```
void initSpeechRecognizer() {
    speechRecognition = SpeechRecognition();

    speechRecognition.setAvailabilityHandler(
      (bool result) => isAvailable = result,
    );

    speechRecognition.setRecognitionResultHandler((String speech) {
      if (speech.length > 0) {
        resultText = speech;
      }
    });

    speechRecognition.setRecognitionCompleteHandler(() {
      isListening = false;
      if (resultText.length > 0) _chatController.senMsg(resultText);
      resultText = '';
    });

    speechRecognition.activate().then(
          (result) => isAvailable = result,
        );
  }
```

.
```
            onTap: () {
              if (!microfoneController.isListening) {
                microfoneController.isListening = true;
                microfoneController.speechRecognition
                    .listen(locale: "pt_BR")
                    .then((result) => print('$result'));
              } else {
                cancelRecording();
              }
            },

```

**Comment by naiadelima** [comment]
I'd also like to get a same solution to this issues =)

**Comment by Edwin-Luijten** [comment]
Encountering the same issue.

**Comment by tan-yaka** [comment]
I had to comment out `speechChannel.invokeMethod("speech.onRecognitionComplete", transcription);` at line 121 of ` android/src/main/java/bz/rxla/flutter/speechrecognition/SpeechRecognitionPlugin.java` to stop it being called twice.

Also commented out `Log.d(LOG_TAG, "onRmsChanged : " + rmsdB);` to cut down on log output.

**Comment by toshiossada** [comment]
How do I do this?
![image](https://user-images.githubusercontent.com/2637049/70332387-3a0a6000-1820-11ea-9842-6f88471a7495.png)

**Comment by tan-yaka** [comment]
Fork the repo and make the changes in your own copy.

You can reference your version in pubspec.yaml with:
```
  speech_recognition:
    git:
      url: git://github.com/{your_github_username}/speech_recognition.git
      ref: master
```

**Comment by toshiossada** [comment]
thanks I'll do this

**Comment by awilliams684** [comment]
So per the two issues on calling completion handler twice I commented the proper lines, in IOS it worked great. However in android I was still receiving multiple callbacks. 

To solve this I added a "hack" in the flutter code in my project to set the speech object to null when the onRecognitionComplete() handler is called. Then next time speechStart() is called I reinitialize.  

void speechStart() {
  print('_speech.speechStart...');
  if (!_isListening) {
    if(_speech == null){
      _speech = SpeechRecognition();
    }
    _speech.listen();
  }
}

void onRecognitionComplete(String text) {
  print('_speech.onRecognitionComplete... $text');
  _speech = null;
    if (text.isNotEmpty && !_isListening) {
    dialogflowRequest(text);
  }
}
